examples/example_tf_training_tfexample/main.py

```diff
@@ -16,34 +16,51 @@
 # under the License.
 #
 
+from __future__ import absolute_import
+
+import tensorflow as tf
 from spotify_tensorflow.dataset import Datasets
 from spotify_tensorflow.trainer import Trainer
-import tensorflow as tf
+
+FLAGS = tf.flags.FLAGS
 
 tf.logging.set_verbosity(tf.logging.INFO)
 
 
-def main(_):
+def train(_):
     from examples_utils import get_data_dir
     import tempfile
 
     config = Trainer.get_default_run_config(tempfile.mkdtemp())
 
-    feature_context = Datasets.get_context(get_data_dir("train"))
+    training_data_dir = get_data_dir("train")
+    feature_context = Datasets.get_context(training_data_dir)
+
     (feature_names, label_names) = feature_context.multispec_feature_groups
     features = [tf.feature_column.numeric_column(x) for x in feature_names]
 
     def split_features_label_fn(spec):
-        label = spec.pop(label_names[0])
-        return spec, label
+        # Canned TF's LinearClassifier requires label to be a single integer, Featran gives us
+        # one hot encoding for class, thus we need to convert one hot encoding to single integer
+        labels = tf.concat([[spec.pop(l)] for l in label_names], axis=0)
+        label = tf.argmax(labels, axis=0)
+        # Get the rest of the features out of the spec
+        split_features = {k: spec[k] for k in set(spec.keys()) - set(label_names)}
```

Comment:
i guess in this case we can just return `spec` since it doesn't contain `label_names` anymore?